### PR TITLE
CC-14662 Project code migration

### PR DESCRIPTION
**Pull request by kkicheglovspryker**
Ticket: https://spryker.atlassian.net/browse/CC-14662

### CODE REVIEW

**Review by sadkivskyi — APPROVED** [review]

File: src/SprykerEco/Zed/UnzerGui/Communication/Plugin/MerchantGui/MerchantUnzerFormExpanderPlugin.php [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+     * @param \Symfony\Component\Form\FormBuilderInterface $builder
+     * @param array $options
+     *
+     * @return \Symfony\Component\Form\FormBuilderInterface
+     */
+    public function expand(FormBuilderInterface $builder, array $options): FormBuilderInterface
+    {
+        $this->addMerchantUnzerParticipantField($builder);
+
+        return $builder;
```
Comment (romanhavrylko) [inline src/SprykerEco/Zed/UnzerGui/Communication/Plugin/MerchantGui/MerchantUnzerFormExpanderPlugin.php]: I suggest moving this to a dedicated business model.

File: src/SprykerEco/Zed/UnzerGui/Communication/Plugin/MerchantGui/Tabs/MerchantProfileUnzerFormTabExpanderPlugin.php [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+     *
+     * @api
+     *
+     * @param \Generated\Shared\Transfer\TabsViewTransfer $tabsViewTransfer
+     *
+     * @return \Generated\Shared\Transfer\TabsViewTransfer
+     */
+    public function expand(TabsViewTransfer $tabsViewTransfer): TabsViewTransfer
+    {
+        $tabItemTransfer = (new TabItemTransfer())->setName('unzer')
```
Comment (romanhavrylko) [inline src/SprykerEco/Zed/UnzerGui/Communication/Plugin/MerchantGui/Tabs/MerchantProfileUnzerFormTabExpanderPlugin.php]: The same as above.

**Review by sadkivskyi — APPROVED** [review]

File: codeception.yml [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+
+paths:
+  tests: tests
+  support: .
+  log: tests/_output
+  data: tests/_data
+  envs: tests/_envs
+
+settings:
+  bootstrap: _bootstrap.php
```
Comment (michbeck) [inline codeception.yml]: Can't we remove this here while it's specified on line 24?

File: src/SprykerEco/Zed/UnzerGui/Dependency/UnzerGuiToUnzerFacadeBridge.php [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+{
+    /**
+     * @var \SprykerEco\Zed\Unzer\Business\UnzerFacadeInterface
+     */
+    protected $unzerFacade;
+
+    /**
+     * @param \SprykerEco\Zed\Unzer\Business\UnzerFacadeInterface $unzerFacade
+     */
+    public function __construct($unzerFacade)
```
Comment (michbeck) [inline src/SprykerEco/Zed/UnzerGui/Dependency/UnzerGuiToUnzerFacadeBridge.php:22]:
```suggestion
    public function __construct(UnzerFacadeInterface  $unzerFacade)
```

Missing type hint?

**Review by michbeck — COMMENTED** [review]
Please see comments ...

File: src/SprykerEco/Zed/UnzerGui/Dependency/UnzerGuiToUnzerFacadeBridge.php (hunk 4 above)
Comment (kkicheglovspryker) [reply src/SprykerEco/Zed/UnzerGui/Dependency/UnzerGuiToUnzerFacadeBridge.php:22]: No, it is a bridge. https://spryker.atlassian.net/wiki/spaces/RFC/pages/13434886/[ACCEPTED]+RFC%3A+Bridges

File: src/SprykerEco/Zed/UnzerGui/Communication/Expander/MerchantUnzerFormTabExpander.php [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+     *
+     * @api
+     *
+     * @param \Generated\Shared\Transfer\TabsViewTransfer $tabsViewTransfer
+     *
+     * @return \Generated\Shared\Transfer\TabsViewTransfer
+     */
+    public function expand(TabsViewTransfer $tabsViewTransfer): TabsViewTransfer
+    {
+        $tabItemTransfer = (new TabItemTransfer())->setName('unzer')
```
Comment (romanhavrylko) [inline src/SprykerEco/Zed/UnzerGui/Communication/Expander/MerchantUnzerFormTabExpander.php]: Can we move this to constant?

File: src/SprykerEco/Zed/UnzerGui/Communication/Expander/MerchantUnzerFormTabExpanderInterface.php [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+
+use Generated\Shared\Transfer\TabsViewTransfer;
+
+interface MerchantUnzerFormTabExpanderInterface
+{
+    /**
+     * Specification:
+     * - Expands tabs for Merchant form.
+     *
+     * @api
```
Comment (romanhavrylko) [inline src/SprykerEco/Zed/UnzerGui/Communication/Expander/MerchantUnzerFormTabExpanderInterface.php]: It's not a public API.

File: src/SprykerEco/Zed/UnzerGui/Communication/Expander/MerchantUnzerFormTabExpander.php [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+{
+    /**
+     * @var string
+     */
+    protected const UNZER_TAB_TEMPLATE = '@UnzerGui/_partials/unzer-tab.twig';
+
+    /**
+     * {@inheritDoc}
+     *
+     * @api
```
Comment (romanhavrylko) [inline src/SprykerEco/Zed/UnzerGui/Communication/Expander/MerchantUnzerFormTabExpander.php]: It's not a public API.

File: src/SprykerEco/Zed/UnzerGui/Communication/Expander/MerchantUnzerFormExpander.php [hunk 8]
```diff
@@ -0,0 +1,79 @@
...
+     */
+    public function __construct(MerchantUnzerFormDataProvider $merchantUnzerFormDataProvider)
+    {
+        $this->merchantUnzerDataProvider = $merchantUnzerFormDataProvider;
+    }
+
+    /**
+     * {@inheritDoc}
+     *
+     * @api
```
Comment (romanhavrylko) [inline src/SprykerEco/Zed/UnzerGui/Communication/Expander/MerchantUnzerFormExpander.php]: It's not a public API.

File: src/SprykerEco/Zed/UnzerGui/Communication/Expander/MerchantUnzerFormExpanderInterface.php [hunk 9]
```diff
@@ -0,0 +1,26 @@
...
+
+use Symfony\Component\Form\FormBuilderInterface;
+
+interface MerchantUnzerFormExpanderInterface
+{
+    /**
+     * Specification:
+     * - Expands Merchant form.
+     *
+     * @api
```
Comment (romanhavrylko) [inline src/SprykerEco/Zed/UnzerGui/Communication/Expander/MerchantUnzerFormExpanderInterface.php]: It's not a public API.

**Review by romanhavrylko — APPROVED** [review]

File: .scrutinizer.yml [hunk 10]
```diff
@@ -0,0 +1,24 @@
+build:
+  environment:
+    php: '7.3'
```
Comment (dmytro-sokolovskyi) [inline .scrutinizer.yml]: Use `7.4` version

File: README.md [hunk 11]
```diff
@@ -0,0 +1,16 @@
+# UnzerGui Module
+[![CI](https://github.com/spryker-eco/unzer-gui/actions/workflows/ci.yml/badge.svg)](https://github.com/spryker-eco/unzer-gui/actions/workflows/ci.yml)
+[![Latest Stable Version](https://poser.pugx.org/spryker-eco/unzer-gui/v/stable.svg)](https://packagist.org/packages/spryker-eco/unzer-gui)
+[![License](https://img.shields.io/github/license/spryker-eco/unzer-gui.svg?b=master)](https://github.com/spryker-eco/unzer-gui)
+[![Scrutinizer Code Quality](https://scrutinizer-ci.com/g/spryker-eco/unzer-gui/badges/quality-score.png?b=master)](https://scrutinizer-ci.com/g/spryker-eco/unzer-gui/?branch=master)
+[![Build Status](https://scrutinizer-ci.com/g/spryker-eco/unzer-gui/badges/build.png?b=master)](https://scrutinizer-ci.com/g/spryker-eco/unzer-gui/build-status/master)
+[![Minimum PHP Version](https://img.shields.io/badge/php-%3E%3D%207.3-8892BF.svg)](https://php.net/)
```
Comment (dmytro-sokolovskyi) [inline README.md]: `https://img.shields.io/badge/php-%3E%3D%207.3-8892BF.svg` -> `https://img.shields.io/badge/php-%3E%3D%207.4-8892BF.svg`

File: composer.json [hunk 12]
```diff
@@ -0,0 +1,40 @@
+{
+    "name": "spryker-eco/unzer-gui",
+    "type": "library",
+    "description": "Unzer GUI module",
+    "license": "MIT",
+    "require": {
+        "php": ">=7.3",
```
Comment (dmytro-sokolovskyi) [inline composer.json]: Use` 7.4` version.

File: src/SprykerEco/Zed/UnzerGui/UnzerGuiDependencyProvider.php [hunk 13]
```diff
@@ -0,0 +1,48 @@
...
+    /**
+     * @param \Spryker\Zed\Kernel\Container $container
+     *
+     * @return \Spryker\Zed\Kernel\Container
+     */
+    public function provideCommunicationLayerDependencies(Container $container): Container
+    {
+        $container = parent::provideCommunicationLayerDependencies($container);
+
+        $container = $this->addUnzerFacade($container);
```
Comment (dmytro-sokolovskyi) [inline src/SprykerEco/Zed/UnzerGui/UnzerGuiDependencyProvider.php:30]: Please, add `spryker-eco/unzer` dependency into `composer.json`

File: src/SprykerEco/Zed/UnzerGui/Communication/Expander/MerchantUnzerFormExpander.php [hunk 14]
```diff
@@ -0,0 +1,75 @@
...
+    {
+        $builder->add(static::FIELD_MERCHANT_UNZER_PARTICIPANT_ID, TextType::class, $options);
+
+        return $builder;
+    }
+
+    /**
+     * @param \Symfony\Component\Form\FormBuilderInterface $builder
+     *
+     * @return string[]
```
Comment (dmytro-sokolovskyi) [inline src/SprykerEco/Zed/UnzerGui/Communication/Expander/MerchantUnzerFormExpander.php]: `string[]` -> `array<string>`

File: src/SprykerEco/Zed/UnzerGui/Communication/Form/DataProvider/MerchantUnzerFormDataProvider.php [hunk 15]
```diff
@@ -0,0 +1,49 @@
...
+     */
+    public function getData(MerchantTransfer $merchantTransfer): ?string
+    {
+        $unzerMerchant = $this->unzerFacade->getUnzerMerchantByMerchantReference($merchantTransfer->getMerchantReference());
+
+        return $unzerMerchant->getParticipantId();
+    }
+
+    /**
+     * @return string[]
```
Comment (dmytro-sokolovskyi) [inline src/SprykerEco/Zed/UnzerGui/Communication/Form/DataProvider/MerchantUnzerFormDataProvider.php]: `string[]` -> `array<string>`

File: src/SprykerEco/Zed/UnzerGui/Communication/Plugin/MerchantGui/MerchantUnzerFormExpanderPlugin.php [hunk 16]
```diff
@@ -0,0 +1,33 @@
...
+
+/**
+ * MIT License
+ * For full license information, please view the LICENSE file that was distributed with this source code.
+ */
+
+namespace SprykerEco\Zed\UnzerGui\Communication\Plugin\MerchantGui;
+
+use Spryker\Zed\Kernel\Communication\AbstractPlugin;
+use Spryker\Zed\MerchantGuiExtension\Dependency\Plugin\MerchantFormExpanderPluginInterface;
```
Comment (dmytro-sokolovskyi) [inline src/SprykerEco/Zed/UnzerGui/Communication/Plugin/MerchantGui/MerchantUnzerFormExpanderPlugin.php]: Please, add `spryker/merchant-gui-extension` dependency into `composer.json`

File: src/SprykerEco/Zed/UnzerGui/Communication/Plugin/MerchantGui/MerchantUnzerFormExpanderPlugin.php [hunk 17]
```diff
@@ -0,0 +1,33 @@
...
+use Symfony\Component\Form\FormBuilderInterface;
+
+/**
+ * @method \SprykerEco\Zed\UnzerGui\Communication\UnzerGuiCommunicationFactory getFactory()
+ */
+class MerchantUnzerFormExpanderPlugin extends AbstractPlugin implements MerchantFormExpanderPluginInterface
+{
+    /**
+     * {@inheritDoc}
```
Comment (dmytro-sokolovskyi) [inline src/SprykerEco/Zed/UnzerGui/Communication/Plugin/MerchantGui/MerchantUnzerFormExpanderPlugin.php]: Please, add description for this plugin.

File: src/SprykerEco/Zed/UnzerGui/Communication/Plugin/MerchantGui/Tabs/MerchantProfileUnzerFormTabExpanderPlugin.php [hunk 18]
```diff
@@ -0,0 +1,32 @@
...
+use Spryker\Zed\Kernel\Communication\AbstractPlugin;
+use Spryker\Zed\MerchantGuiExtension\Dependency\Plugin\MerchantFormTabExpanderPluginInterface;
+
+/**
+ * @method \SprykerEco\Zed\UnzerGui\Communication\UnzerGuiCommunicationFactory getFactory()
+ */
+class MerchantProfileUnzerFormTabExpanderPlugin extends AbstractPlugin implements MerchantFormTabExpanderPluginInterface
+{
+    /**
+     * {@inheritDoc}
```
Comment (dmytro-sokolovskyi) [inline src/SprykerEco/Zed/UnzerGui/Communication/Plugin/MerchantGui/Tabs/MerchantProfileUnzerFormTabExpanderPlugin.php]: Please, add description for this plugin.

File: src/SprykerEco/Zed/UnzerGui/Communication/Plugin/MerchantGui/Tabs/MerchantProfileUnzerFormTabExpanderPlugin.php [hunk 19]
```diff
@@ -0,0 +1,32 @@
...
+namespace SprykerEco\Zed\UnzerGui\Communication\Plugin\MerchantGui\Tabs;
+
+use Generated\Shared\Transfer\TabsViewTransfer;
+use Spryker\Zed\Kernel\Communication\AbstractPlugin;
+use Spryker\Zed\MerchantGuiExtension\Dependency\Plugin\MerchantFormTabExpanderPluginInterface;
+
+/**
+ * @method \SprykerEco\Zed\UnzerGui\Communication\UnzerGuiCommunicationFactory getFactory()
+ */
+class MerchantProfileUnzerFormTabExpanderPlugin extends AbstractPlugin implements MerchantFormTabExpanderPluginInterface
```
Comment (dmytro-sokolovskyi) [inline src/SprykerEco/Zed/UnzerGui/Communication/Plugin/MerchantGui/Tabs/MerchantProfileUnzerFormTabExpanderPlugin.php]: Plugin names should have a suffix of plugin interface name, 
e. g. `MerchantFormTabExpanderPluginInterface` → `[MerchantProfileUnzer]MerchantFormTabExpanderPlugin` or `[ProfileUnzer]MerchantFormTabExpanderPlugin`

File: src/SprykerEco/Zed/UnzerGui/Communication/Plugin/MerchantGui/MerchantUnzerFormExpanderPlugin.php [hunk 20]
```diff
@@ -0,0 +1,33 @@
...
+namespace SprykerEco\Zed\UnzerGui\Communication\Plugin\MerchantGui;
+
+use Spryker\Zed\Kernel\Communication\AbstractPlugin;
+use Spryker\Zed\MerchantGuiExtension\Dependency\Plugin\MerchantFormExpanderPluginInterface;
+use Symfony\Component\Form\FormBuilderInterface;
+
+/**
+ * @method \SprykerEco\Zed\UnzerGui\Communication\UnzerGuiCommunicationFactory getFactory()
+ */
+class MerchantUnzerFormExpanderPlugin extends AbstractPlugin implements MerchantFormExpanderPluginInterface
```
Comment (dmytro-sokolovskyi) [inline src/SprykerEco/Zed/UnzerGui/Communication/Plugin/MerchantGui/MerchantUnzerFormExpanderPlugin.php]: Plugin names should have a suffix of plugin interface name, 
e. g. `MerchantFormExpanderPluginInterface ` → `[Unzer]MerchantFormExpanderPlugin`.

File: .scrutinizer.yml (hunk 10 above)
Comment (michbeck) [reply .scrutinizer.yml]: Fixed

File: README.md (hunk 11 above)
Comment (michbeck) [reply README.md]: Fixed

File: composer.json (hunk 12 above)
Comment (michbeck) [reply composer.json]: Fixed

File: src/SprykerEco/Zed/UnzerGui/UnzerGuiDependencyProvider.php (hunk 13 above)
Comment (michbeck) [reply src/SprykerEco/Zed/UnzerGui/UnzerGuiDependencyProvider.php:30]: Fixed

File: src/SprykerEco/Zed/UnzerGui/Communication/Expander/MerchantUnzerFormExpander.php (hunk 14 above)
Comment (michbeck) [reply src/SprykerEco/Zed/UnzerGui/Communication/Expander/MerchantUnzerFormExpander.php]: Fixed

File: src/SprykerEco/Zed/UnzerGui/Communication/Form/DataProvider/MerchantUnzerFormDataProvider.php (hunk 15 above)
Comment (michbeck) [reply src/SprykerEco/Zed/UnzerGui/Communication/Form/DataProvider/MerchantUnzerFormDataProvider.php]: @dmytro-sokolovskyi Shouldn't it be array<string,string>?

File: src/SprykerEco/Zed/UnzerGui/Communication/Plugin/MerchantGui/MerchantUnzerFormExpanderPlugin.php (hunk 16 above)
Comment (michbeck) [reply src/SprykerEco/Zed/UnzerGui/Communication/Plugin/MerchantGui/MerchantUnzerFormExpanderPlugin.php]: Fixed

File: src/SprykerEco/Zed/UnzerGui/Communication/Form/DataProvider/MerchantUnzerFormDataProvider.php (hunk 15 above)
Comment (michbeck) [reply src/SprykerEco/Zed/UnzerGui/Communication/Form/DataProvider/MerchantUnzerFormDataProvider.php]: Fixed

File: src/SprykerEco/Zed/UnzerGui/Communication/Expander/MerchantUnzerFormTabExpander.php (hunk 5 above)
Comment (michbeck) [reply src/SprykerEco/Zed/UnzerGui/Communication/Expander/MerchantUnzerFormTabExpander.php]: Fixed

File: src/SprykerEco/Zed/UnzerGui/Communication/Expander/MerchantUnzerFormTabExpanderInterface.php (hunk 6 above)
Comment (michbeck) [reply src/SprykerEco/Zed/UnzerGui/Communication/Expander/MerchantUnzerFormTabExpanderInterface.php]: Fixed

File: src/SprykerEco/Zed/UnzerGui/Communication/Expander/MerchantUnzerFormTabExpander.php (hunk 7 above)
Comment (michbeck) [reply src/SprykerEco/Zed/UnzerGui/Communication/Expander/MerchantUnzerFormTabExpander.php]: Fixed

File: src/SprykerEco/Zed/UnzerGui/Communication/Expander/MerchantUnzerFormExpanderInterface.php (hunk 9 above)
Comment (michbeck) [reply src/SprykerEco/Zed/UnzerGui/Communication/Expander/MerchantUnzerFormExpanderInterface.php]: Fixed

File: src/SprykerEco/Zed/UnzerGui/Communication/Plugin/MerchantGui/MerchantUnzerFormExpanderPlugin.php (hunk 17 above)
Comment (michbeck) [reply src/SprykerEco/Zed/UnzerGui/Communication/Plugin/MerchantGui/MerchantUnzerFormExpanderPlugin.php]: Fixed

File: src/SprykerEco/Zed/UnzerGui/Communication/Plugin/MerchantGui/Tabs/MerchantProfileUnzerFormTabExpanderPlugin.php (hunk 18 above)
Comment (michbeck) [reply src/SprykerEco/Zed/UnzerGui/Communication/Plugin/MerchantGui/Tabs/MerchantProfileUnzerFormTabExpanderPlugin.php]: Fixed

File: src/SprykerEco/Zed/UnzerGui/Communication/Plugin/MerchantGui/MerchantUnzerFormExpanderPlugin.php (hunk 20 above)
Comment (michbeck) [reply src/SprykerEco/Zed/UnzerGui/Communication/Plugin/MerchantGui/MerchantUnzerFormExpanderPlugin.php]: Fixed

File: src/SprykerEco/Zed/UnzerGui/Communication/Plugin/MerchantGui/Tabs/MerchantProfileUnzerFormTabExpanderPlugin.php (hunk 19 above)
Comment (michbeck) [reply src/SprykerEco/Zed/UnzerGui/Communication/Plugin/MerchantGui/Tabs/MerchantProfileUnzerFormTabExpanderPlugin.php]: Fixed

**Review by dmytro-sokolovskyi — APPROVED** [review]

**Pull request merged**